src/cursor.rs

```diff
@@ -0,0 +1,171 @@
+use core::fmt;
+use std::{error::Error, sync::Arc};
+
+use crate::{icon::PIXEL_SIZE, platform_impl::PlatformCustomCursor};
+
+#[derive(Debug, Clone)]
+pub struct CustomCursor {
+    pub(crate) inner: Arc<PlatformCustomCursor>,
```

Comment:
If we remove `Clone` I think the `Arc` isn't necessary either. Backends which need it, like Web, could add an `Arc` to their `PlatformCustomCursor`.

Reply:
Now `PlatformCustomCursor` includes the rgba buffer, so I don't like copying it every time the user want's to set the same cursor multiple times just like in my example. `Clone` is necessary and it would be nice for it to be cheap.

Reply:
But I guess the user could also put it in an arc themselves. So whatever works for me.

Reply:
See https://github.com/rust-windowing/winit/pull/3218#discussion_r1397911756.
My thought was if `Window::set_custom_cursor()` takes a reference, users don't have to clone and can keep it.

The assumption here is that backends actually only need a reference, right?

Reply:
My comment on the thread above indicates that arc is still needed.

Reply:
For documentation purpose so we can decide that later:
if we use `maybe_wait_on_main()` in `Window::set_custom_cursor()` we can remove the `Arc` if we want to. But I don't see a reason to do that right now considering `maybe_wait_on_main()` is strictly worse then `maybe_queue_on_main()` for the user.